Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -686,6 +686,12 @@ namespace Babylon
         auto uniforms = Napi::Array::New(info.Env(), length);
         for (uint32_t index = 0; index < length; ++index)
         {
+            if (!names[index].IsString())
+            {
+                uniforms[index] = info.Env().Null();
+                continue;
+            }
+
             const auto name = names[index].As<Napi::String>().Utf8Value();
```

Comment:
What you have seems fine. Having it assign null values in two separate places, one which is effectively an early-out with a `continue` and one which is a fallthrough case, seems maybe a _little_ weird, so I personally might guard against it by just giving non-string queries an impossible name, allowing them to fall through to the existing handler. That's probably more a stylistic choice than anything else, though, as I think both approaches should work.

```suggestion
            const auto name = names[index].IsString() ? names[index].As<Napi::String>().Utf8Value() : "";
```